hwy/tests/memory_test.cc

```diff
@@ -324,6 +324,254 @@ HWY_NOINLINE void TestAllCache() {
   Pause();
 }
 
+class TestLoadN {
+ private:
+  template <class T, HWY_IF_NOT_FLOAT_NOR_SPECIAL(T)>
+  static HWY_INLINE T GenerateNonZeroValue(size_t val) {
+    const T conv_val = static_cast<T>(val);
+    return (conv_val == 0) ? static_cast<T>(-17) : conv_val;
+  }
+  template <class T, HWY_IF_FLOAT3264(T)>
+  static HWY_INLINE T GenerateNonZeroValue(size_t val) {
+    const T flt_val = static_cast<T>(val);
+    return (flt_val == T{0} ? static_cast<T>(0.5426808228865735) : flt_val);
+  }
+  template <class T, hwy::EnableIf<IsSame<T, hwy::bfloat16_t>()>* = nullptr>
+  static HWY_INLINE T GenerateNonZeroValue(size_t val) {
+    return BF16FromF32(GenerateNonZeroValue<float>(val));
+  }
+  template <class T, hwy::EnableIf<IsSame<T, hwy::float16_t>()>* = nullptr>
+  static HWY_INLINE T GenerateNonZeroValue(size_t val) {
+    return F16FromF32(GenerateNonZeroValue<float>(val));
+  }
+
+ public:
+  template <class T, class D>
+  HWY_NOINLINE void operator()(T /*unused*/, D d) {
+    const size_t N = Lanes(d);
+    constexpr size_t kMaxLanesPerBlock = 16 / sizeof(T);
+    const size_t lpb = HWY_MIN(N, kMaxLanesPerBlock);
+    HWY_ASSERT(lpb >= 1);
+    HWY_ASSERT(N <= (static_cast<size_t>(~size_t(0)) / 4));
+
+    const size_t load_buf_len = (3 * N) + 4;
+
+    auto load_buf = AllocateAligned<T>(load_buf_len);
+    auto expected = AllocateAligned<T>(N);
+    HWY_ASSERT(load_buf && expected);
+
+    for (size_t i = 0; i < load_buf_len; i++) {
+      load_buf[i] = GenerateNonZeroValue<T>(i + 1);
+    }
+
+    std::fill(expected.get(), expected.get() + N, T(0));
```

Comment:
We have problems with libc++ on some platforms. In future, let's use ZeroBytes instead. No need to change now, I will replace it when it becomes necessary.